app/src/main/java/com/github/se/icebreakrr/MainActivity.kt

```diff
@@ -92,14 +93,20 @@ fun IcebreakrrApp(auth: FirebaseAuth, functions: FirebaseFunctions) {
   val profileViewModel: ProfilesViewModel = viewModel(factory = ProfilesViewModel.Factory)
   val tagsViewModel: TagsViewModel = viewModel(factory = TagsViewModel.Factory)
   val filterViewModel: FilterViewModel = viewModel(factory = FilterViewModel.Factory)
-  val ourUserUid = auth.currentUser?.uid ?: "null"
-  val ourName = auth.currentUser?.displayName ?: "null"
+  var userName: String? = "null"
+  var userUid: String? = "null"
+  auth.currentUser?.reload()?.addOnCompleteListener { task ->
+    userName = auth.currentUser?.displayName
+    userUid = auth.currentUser?.uid
+  }
+  Log.d("USER ID", userUid ?: "null")
+  Log.d("DISPLAY NAME", userName ?: "null")
```

Comment:
I don't understand the upsides of using reload instead of directly taking the current user.
You can also get rid of the debugging logs

Reply:
It was the problem i had, when i had just currentUser, i had sometimes the user returned as "null", the reload guarantees that we get the good user name at all times

Reply:
Thank you for the clarification!